### DISCUSSION TITLE
Viewport previewing

### DISCUSSION THREAD

**Discussion by chrisvxd** [puckeditor/puck#72] [Ideas]
Users may wish to preview their page for different viewport sizes without leaving Puck.

Viewport previews are typically done by resizing an iframe containing the content. However, Puck doesn't currently use frames because of their historical performance issues and the challenges supporting a robust drag-and-drop interface.

We're looking for alternative solutions. Please use this thread to propose ideas and discuss!

**Comment by chrisvxd** [comment]
Proposal: a "full screen" icon button that makes the preview window full screen and disables any zoom, allowing the user to resize their window appropriately.

**Reply by chrisvxd** [reply]
Alternatively, we can provide viewport previewing buttons and open a restricted preview in an iframe modal, without any editing functionality.

**Reply by jswhisperer** [reply]
An outside the box (frame) idea https://html2canvas.hertzen.com/

**Comment by chrisvxd** [comment]
Possible as of #353
